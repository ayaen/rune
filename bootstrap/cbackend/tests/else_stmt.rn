//  Copyright 2024 Google LLC.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

if false {
  println "never"
} else {
  println "space is the place"
}
if false {
  println "never"
} else {
  // empty block
}
if true {
  println "no else this time!"
} else {
  println "never"
}
a = 1
b = 2
if a + b != 3 {
  println "never"
} else {
  println "a + b == 3"
}
if a + b - 1 >= 3 {
  println "never"
} else {
  println "a + b - 1 < 3"
}
herp = "herp"
derp = "derp"
if derp == herp {
  println "never"
} else {
  println herp, " ", derp
}