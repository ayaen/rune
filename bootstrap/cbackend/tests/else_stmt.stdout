space is the place
no else this time!
a + b == 3
a + b - 1 < 3
herp derp
